generate/download.go

```diff
@@ -175,6 +175,12 @@ func githubAssets(repo, version string) (string, []githubAsset, error) {
 	}
 	defer res.Body.Close()
 
+	if res.StatusCode >= 400 {
+		// This can easily happen if we make tons of requests to GitHub
+		// from one machine, as the IP hits their default rate limit.
+		return "", nil, fmt.Errorf("GET %s: %s", urlstr, http.StatusText(res.StatusCode))
```

Comment:
perhaps:

```
       // see https://developer.github.com/v3/#rate-limiting
	var apiError struct {
		Message string `json:"message"`
	}
	if err := json.NewDecoder(res.Body).Decode(&apiError); err != nil {
		return "", nil, fmt.Errorf("decoding error response: %v", err)
	}
 	return "", nil, fmt.Errorf("GET %s: %s: %s", urlstr, http.StatusText(res.StatusCode), apiError.Message) 
```
or similar?

`error: downloading protoc: GET https://api.github.com/repos/ProtocolBuffers/protobuf/releases/latest: Forbidden` is still a little opaque

Reply:
*shrug* I thought about it, but at the same time I want to get rid of this code soon, so I thought it wasn't worth the effort. At that point we might as well use a github API client.